File: server_JAVA/src/main/java/com/example/demo/models/RecipeRating.java
package com.example.demo.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIncludeProperties;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;

@Entity
@Table(name = "recipe_rating")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class RecipeRating {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "recipe_id", nullable = false)
    @JsonIgnore
    private Recipe recipe;

    @ManyToOne
    @JoinColumn(name = "author_id", nullable = false)
    @JsonIncludeProperties({"id"})
    private User author;

    @Column(name = "type", nullable = false)
    private String type;

    @Column(name = "created_at")
    private Timestamp createdAt = new Timestamp(System.currentTimeMillis());

    @Column(name = "updated_at")
    private Timestamp updatedAt;
}

File: server/src/main/kotlin/com/example/demo/recipe/domain/reciperating/RecipeRating.kt
package com.example.demo.recipe.domain.reciperating

import com.example.demo.user.domain.Recipe
import com.example.demo.user.domain.User

data class RecipeRating(
    var id: Int = 0,
    var type: RecipeRatingType,
    var recipe: Recipe,
    var author: User,
)

/*
@Entity
@Table(
    name = "recipe_ratings",
    uniqueConstraints = [UniqueConstraint(columnNames = ["recipe_id", "user_id"])],
)
data class RecipeRating(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) var id: Int = 0,
    @Enumerated(EnumType.STRING)
    @Column(name = "type", nullable = false)
    var type: RecipeRatingType,
    @ManyToOne @JoinColumn(name = "recipe_id", nullable = false) var recipe: Recipe,
    @ManyToOne @JoinColumn(name = "user_id", nullable = false) var author: User,
)
*/
